cmake/generic.cmake

```diff
@@ -331,3 +331,52 @@ function(go_test TARGET_NAME)
   add_custom_target(${TARGET_NAME} ALL DEPENDS ${TARGET_NAME}_timestamp ${go_test_DEPS})
   add_test(${TARGET_NAME} ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME})
 endfunction(go_test)
+
+function(pb_cc_library TARGET_NAME)
```

Comment:
I have tested the use of `protobuf_generate_cpp` in #2582 and found out the standard function `prptobuf_generate_cpp` is not good enough to use because of two reasons:
1. Function `protobuf_generate_cpp` is defined in cmake's `Protobuf.cmake`. In the case that `find_package(Protobuf)` is not called, the build process will be aborted because of `Unknown CMake command "protobuf_generate_cpp"`  error.
2. In `protobuf_generate_cpp`, we cannot set the dependency of others, and the function may be called before the `protoc` executable in third_party generated.